### DISCUSSION TITLE
Negation detection as a Multi-task learning (MTL) layer

### DISCUSSION THREAD

**Discussion by nipunsadvilkar** [explosion/spaCy#4131] [New Features & Project Ideas]
I have been going through this paper - [Joint Entity Extraction and Assertion Detection for Clinical Text](https://arxiv.org/pdf/1812.05270.pdf) - which proposes an MTL approach to negation detection that leverages overlapping representation across sub-tasks i.e., jointly model named entity and negation
in an end-to-end system.

I already have NER model in place and was thinking how would I implement MTL within spacy. Through some research, I see @honnibal 	has already done some work and provided an example - [ner_multitask_objective.py]( https://github.com/explosion/spaCy/blob/master/examples/training/ner_multitask_objective.py) but I find it difficult mold above example into negation multitask.

@honnibal @ines : Would love to know your take on how to go about it and would like to contribute to repo if it pans out well.

**Comment by p-sodmann** [comment]
I am currently working on NER and Negation detection (3 layer) bilstm network on clinical notes in german. It might be easier to do the learning part in keras / pytorch since the documentation is much more complete and there are more resources on how to implement these algorithms.

**Comment by svlandeg** [comment]
I think many use-cases would benefit from this kind of negation detection system, and it would be a useful addition to existing components in spaCy. At the moment, it's probably best implemented as a external module. There's already an extension available from the [universe](https://spacy.io/universe): https://github.com/jenojp/negspacy. If this one doesn't suit your needs, and you end up implementing a different model, feel free to submit it to the spaCy universe as well!
